extended/src/main/java/io/kubernetes/client/extended/workqueue/ratelimiter/RateLimiter.java

```diff
@@ -0,0 +1,23 @@
+package io.kubernetes.client.extended.workqueue.ratelimiter;
+
+import java.time.Duration;
+
+public interface RateLimiter<T> {
```

Comment:
Why not use RateLimiter from Guava?

https://github.com/google/guava/blob/master/guava/src/com/google/common/util/concurrent/RateLimiter.java

Reply:
the guava rate-limiter's API is different from the one we want for work-queue here i suppose. guava's rate-limiter is basically expecting an invocation to be blocked until permitted, while for work-queue, the so-called `RateLimiter` here is more like a backoff-provider, it will return how much time an inbound item need to be waiting for next round of processing.. probably we should consider rename this to sth like `BackOffGetter`, or `BackOffFactory`? @brendandburns @cizezsy wdyt?

Reply:
Yeah, I have looked at Guava RateLimiter before, but it does not satisfy our requirement. 

Regard the name, maybe we should keep the name same with client-go's so that others can quickly know how to use client-java?